packages/apps/job-launcher/server/src/modules/payment/currency.service.ts

```diff
@@ -1,25 +1,31 @@
 import { HttpService } from '@nestjs/axios';
-import { Injectable, NotFoundException } from '@nestjs/common';
+import { Injectable, Logger, NotFoundException } from '@nestjs/common';
 import { firstValueFrom } from 'rxjs';
 import { COINGECKO_API_URL } from '../../common/constants';
 import { Currency, TokenId } from '../../common/enums/payment';
 import { ErrorCurrency } from '../../common/constants/errors';
 
 @Injectable()
 export class CurrencyService {
+  private readonly logger = new Logger(HttpService.name);
   constructor(private readonly httpService: HttpService) {}
 
   public async getRate(tokenId: TokenId, currency: Currency): Promise<number> {
-    const { data } = await firstValueFrom(
-      await this.httpService.get(
-        `${COINGECKO_API_URL}?ids=${tokenId}&vs_currencies=${currency}`,
-      ),
-    );
+    try {
```

Comment:
move this try catch to the controller

Reply:
Can you explain? I think that we should follow the rule - thin controllers and keep all the logic and error handling at the service level. There is no particular point in moving the error trapping block to the controller level. This `getRate` method will be used in other services, I suggest leaving this `try...catch` block where it is or removing it, but it cannot be in the controller.

Reply:
Imagine a situation where error handling requires an if-else condition, and when we have a lot of cases (example: https://github.com/humanprotocol/human-protocol/blob/f3f2de2c38f512eb2f0ba122b236e23a8c4a2e37/packages/sdk/typescript/human-protocol-sdk/src/escrow.ts#L138), we will write error handling logic in the controller, which is not very good.

I propose to completely remove this block, and not to take it out anywhere, since if this method is called in the controller, the error will be caught by the validation pipe. If this method is called from another service, it can be wrapped in a try-catch.